### PR TITLE
Fallback section text updated

### PR DESCRIPTION
**Pull request by veselinrak**

### CODE REVIEW

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 1]
```diff
@@ -1013,20 +1013,38 @@ handshake is as follows:
 
 When a subflow fails to operate within the MP-DCCP requirements, it is 
 necessary to fall back to the safe operation. This may be either falling back 
-to regular DCCP, or removing a problematic subflow. The main reason for 
-subflow failing is loss of MP-DCCP options. 
+to regular DCCP, or removing a problematic subflow. The main reasons for 
+subflow failing include: loss of MP-DCCP options, failure to negotiate protocol 
+version, or modification of payload data.
```
Comment (markusa) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1018]: What could be problematic payload data which let MP-DCCP fail?

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 2]
```diff
@@ -1013,20 +1013,38 @@ handshake is as follows:
...
+established and the handshake should fall back to regular DCCP. 
+
+The same fallback should take place if the endpoints fail to agree on a protocol
+version to use during the Multipath Capable feature negotiation, which is described
+in section 3.1. The protocol version negotiation distinguishes between negotiation
+for the initial connection establishment, and addition of subsequent subflows. If
+protocol version negotiation is not successful during the initial connection establishment, MP-DCCP connection will fall back
+to regular DCCP. Any subflow addition to an existing MP connection MUST use the
+same version negotiated for the first flow. If a new subflow is not able to use the
+negotiated protocol version, that subflow will be closed.
```
Comment (markusa) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1033]: There is no version negotiation for subsequent flows created with MP_JOIN. So from my view this text is not required.

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 3]
```diff
@@ -1013,20 +1013,38 @@ handshake is as follows:
...
+Another relevant case is when payload data is modified by middleboxes. DCCP uses 
+checksum to protest the data, as described in section 9 of RFC4340. A checksum will 
+fail if the data has been changed in any way. This will also detect if the data 
+sequence mapping is no longer valid. All data from the start of the segment that
+failed the checksum onwards cannot be considered trustworthy. DCCP defines that if 
+the checksum fails, the receiving endpoint MUST drop the application data and report 
+that data as dropped due to corruption using a Data Dropped option (Drop Code 3, 
+Corrupt). If this happens in an MP-DCCP connection, as it is not possible to 
+determine what changes have been made to the data, the affected subflow must be 
+immediately closed using a regular Close or Reset.
```
[the remark this answers is not in the document]
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1047]: As discussed it should be left to the implementor to decider wether to close a subflow or not after receiving corrupted data or the Data Dropped option.

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 4]
```diff
@@ -1013,20 +1013,38 @@ handshake is as follows:
...
-to regular DCCP, or removing a problematic subflow. The main reason for 
-subflow failing is loss of MP-DCCP options. 
+to regular DCCP, or removing a problematic subflow. The main reasons for 
+subflow failing include: loss of MP-DCCP options, failure to negotiate protocol 
+version, or modification of payload data. 
 
 At the start of the MP-DCCP connection, the handshake ensures exchange of MP-DCCP
 options and thus ensures that the path is fully MP-DCCP capable. If during the
 handshake procedure it appears that DCCP-Request or DCCP-Response or DCCP-Ack
 messages don’t have the MP-DCCP options, the MP-DCCP connection will not be
```
Comment (NathalieRM) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1023]: I think the distinction between DCCP feature and MP-DCCP options has to be clear here as well, MP_CAPABLE is defined as a DCCP feature not as MP-DCCP option, the only MP-DCCP option carried in the initial Request, Response and Ack messages is MP_KEY. Based on that, I think the fallback to DCCP should only happen when the MP_CAPABLE feature is not present.
```suggestion
handshake procedure it appears that the DCCP-Request or the DCCP-Response messages don't carry the MP_CAPABLE feature, the MP-DCCP connection will not be 
```

Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1023]: @NathalieRM 
The difference between feature and option I already tried to fix in my suggestion [above](https://github.com/markusa/ietf-multipath-dccp/pull/78#discussion_r901595200).

I don't tend to agree that we can limit the fallback to the feature negotiation. What happens if the request Key Type is not supported or if the final ACK echoing the wrong Key A/B? What happens if the wrong Path token is used in MP_JOIN? ...

Comment (NathalieRM) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1023]: You are right, it shouldn't happen only if the MP_CAPABLE feature is not present, the point is the text stated initially that the fallback should happen due to the abscence of MP-DCCP options in the initial 3 messages,  and I think it should happen due to the abscence of the MP_CAPABLE feature. The scenarios you mention are not abscence of MP options, and they could have different ways of handling, for the case of not supported key type and echoing wrong key, fallling back to DCCP is fine. For the case of wrong path token in MP_JOIN, I think the establishment of the subflow should just fail, at that point the initial subflow has already been established and the connection is already an MP-DCCP connection, so I don't see the reason to fallback to DCCP

Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1023]: D'accord

Comment (veselinrak) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1023]: Nathalie's comment is not conclusive - 'shouldn't happen only' I read as 'should happen for other reasons, too', although the rest of her comment agrees with Markus's edit above, and does not provide other reasons for fallback. I have therefore made the change according to Markus's edit.

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 5]
```diff
@@ -1013,20 +1013,38 @@ handshake is as follows:
...
 handshake procedure it appears that DCCP-Request or DCCP-Response or DCCP-Ack
 messages don’t have the MP-DCCP options, the MP-DCCP connection will not be 
-established and the handshake should fall back to regular DCCP. The same
-fallback should take place if the endpoints fail to agree on a protocol
-version to use during the Multipath Capable feature negotiation.
+established and the handshake should fall back to regular DCCP. 
+
+The same fallback should take place if the endpoints fail to agree on a protocol
+version to use during the Multipath Capable feature negotiation, which is described
+in section 3.1. The protocol version negotiation distinguishes between negotiation
```
[the remark this answers is not in the document]
Comment (veselinrak) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1028]: done.

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 2 above)
Comment (veselinrak) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1033]: ok. sentence starting 'Any subflow addition...' removed.

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 6]
```diff
@@ -1013,20 +1013,38 @@ handshake is as follows:
...
+for the initial connection establishment, and addition of subsequent subflows. If
+protocol version negotiation is not successful during the initial connection establishment, MP-DCCP connection will fall back
+to regular DCCP. Any subflow addition to an existing MP connection MUST use the
+same version negotiated for the first flow. If a new subflow is not able to use the
+negotiated protocol version, that subflow will be closed. 
 
 If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP options 
-are not present in the handshake messages or the protocol version doesn't
-match the value negotiated for the first flow, that subflow will be closed.
+are not present in the handshake messages, that subflow will be closed.
```
[the remark this answers is not in the document]
Comment (veselinrak) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1036]: changed with small wording modification.

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 7]
```diff
@@ -1013,20 +1013,38 @@ handshake is as follows:
...
+same version negotiated for the first flow. If a new subflow is not able to use the
+negotiated protocol version, that subflow will be closed. 
 
 If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP options 
-are not present in the handshake messages or the protocol version doesn't
-match the value negotiated for the first flow, that subflow will be closed.
+are not present in the handshake messages, that subflow will be closed. 
+
+Another relevant case is when payload data is modified by middleboxes. DCCP uses 
+checksum to protest the data, as described in section 9 of RFC4340. A checksum will
```
[the remark this answers is not in the document]
Comment (veselinrak) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1039]: done

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 3 above)
[the remark this answers is not in the document]
Comment (veselinrak) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1047]: done. this is not left open - there is an option to either close the subflow or to take other action.

[the remark this answers is not in the document]
Comment (veselinrak) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1047]: 'now', not 'not'...

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 8]
```diff
@@ -1013,20 +1013,34 @@ handshake is as follows:
...
+are not present or are faulty in the handshake procedure, that subflow will be closed. 
+
+Another relevant case is when payload data is modified by middleboxes. DCCP uses 
+checksum to protest the data, as described in section 9 of {{RFC4340}}. A checksum will 
+fail if the data has been changed in any way. This will also detect if the data 
+sequence mapping is no longer valid. All data from the start of the segment that
+failed the checksum onwards cannot be considered trustworthy. DCCP defines that if 
+the checksum fails, the receiving endpoint MUST drop the application data and report 
+that data as dropped due to corruption using a Data Dropped option (Drop Code 3, 
+Corrupt). If this happens in an MP-DCCP connection, the affected sublow can either be closed or other action can be taken.
```
Comment (markusa) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1043]:
```suggestion
Corrupt). If this happens in an MP-DCCP connection, the affected subflow can either be closed or other action can be taken. 
```

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 9]
```diff
@@ -1013,20 +1013,34 @@ handshake is as follows:
...
-to regular DCCP, or removing a problematic subflow. The main reason for 
-subflow failing is loss of MP-DCCP options. 
+to regular DCCP, or removing a problematic subflow. The main reasons for 
+subflow failing include: loss of MP-DCCP options, failure to negotiate protocol 
+version, or modification of payload data. 
 
-At the start of the MP-DCCP connection, the handshake ensures exchange of MP-DCCP
+At the start of the MP-DCCP connection, the handshake ensures exchange of MP-DCCP feature and
 options and thus ensures that the path is fully MP-DCCP capable. If during the
 handshake procedure it appears that DCCP-Request or DCCP-Response or DCCP-Ack
```
Comment (NathalieRM) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1022]:
```suggestion
handshake procedure it appears that DCCP-Request or DCCP-Response
```

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 4 above)
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd]: Veselin, I think we also need some considerations what happen if MP Options are missing or faulty as outlined in this thread.

Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd]: Solved in https://github.com/markusa/ietf-multipath-dccp/pull/78/commits/3d0b91dc404d079acf13a9bd8fde0bed1848a576

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 10]
```diff
@@ -1013,20 +1013,41 @@ handshake is as follows:
...
-established and the handshake should fall back to regular DCCP. The same
-fallback should take place if the endpoints fail to agree on a protocol
-version to use during the Multipath Capable feature negotiation.
-
-If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP options 
-are not present in the handshake messages or the protocol version doesn't
-match the value negotiated for the first flow, that subflow will be closed.
+handshake procedure it appears that DCCP-Request or DCCP-Response
+messages don’t carry the MP_CAPABLE feature, the MP-DCCP connection will not be 
+established and the handshake SHOULD fall back to regular DCCP or MUST closed.
```
Comment (NathalieRM) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1025]:
```suggestion
established and the handshake SHOULD fall back to regular DCCP or MUST be closed. 
```

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 11]
```diff
@@ -1013,20 +1013,41 @@ handshake is as follows:
...
+If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP options or MP_CAPABLE
+feature are not present or are faulty in the handshake procedure, that subflow MUST be closed.
+This is especially the case if a different MP_CAPABLE version than the originally negotiated
+version is used. Also non-verifiable MP_HMAC {{MP_HMAC}} or MP_JOIN Path Token {{MP_JOIN}} MUST
+lead to a subflow closing.
+
+Another relevant case is when payload data is modified by middleboxes. DCCP uses 
+checksum to protest the data, as described in section 9 of {{RFC4340}}. A checksum will 
+fail if the data has been changed in any way. This will also detect if the data 
+sequence mapping is no longer valid. All data from the start of the segment that
```
Comment (NathalieRM) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1046]: I'm still confused about -data sequence mapping-, from what I understood in our previuos discussion, this is a feature used by MP-TCP to map both sequence number spaces, but it is not defined for MP-DCCP.... unless I got something wrong.

Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd]: Solved in https://github.com/markusa/ietf-multipath-dccp/pull/78/commits/335375157b490884db3439465bcbcac6c86c8cbc

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 10 above)
Comment (dirkhugo) [inline draft-ietf-tsvwg-multipath-dccp.mkd]: MUST closed => MUST be closed

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 12]
```diff
@@ -1013,20 +1013,41 @@ handshake is as follows:
...
+a final ACK carrying MP_KEY with wrong Key-A/Key-B is received or MP_KEY option is malformed.
+
+If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP options or MP_CAPABLE
+feature are not present or are faulty in the handshake procedure, that subflow MUST be closed.
+This is especially the case if a different MP_CAPABLE version than the originally negotiated
+version is used. Also non-verifiable MP_HMAC {{MP_HMAC}} or MP_JOIN Path Token {{MP_JOIN}} MUST
+lead to a subflow closing.
+
+Another relevant case is when payload data is modified by middleboxes. DCCP uses 
+checksum to protest the data, as described in section 9 of {{RFC4340}}. A checksum will
```
Comment (dirkhugo) [inline draft-ietf-tsvwg-multipath-dccp.mkd]: checksum to protest the data => checksum to protect the data

**Comment by dirkhugo** [comment]
beyond the 2 nits the new text seems to be fine for me - thanks!

**Pull request merged**